hazelcast/src/main/java/com/hazelcast/internal/management/operation/ScriptExecutorOperation.java

```diff
@@ -71,16 +73,18 @@ public Object getResponse() {
     protected void writeInternal(ObjectDataOutput out) throws IOException {
         out.writeUTF(engineName);
         out.writeUTF(script);
-        // kept for compatibility
-        out.writeInt(0);
+        if (out.getVersion().isLessThan(V3_10)) {
+            out.writeInt(0);
```

Comment:
I don't know how `Versioned` or object layout in communication in general works and if it allows to introduce different binary formats depending on the version if this has not been there form the very beginning. 

As I see it different binary formats can only work properly for all members in a cluster if they are aware of versioning from the very beginning so that they read the version as first value of the serialized object. Than they can decide what to do to read a object in a certain version. The above code might be just that, except that the version reading part is hidden in the `ObjectDataOutput` and `ObjectDataInput`.